extmod/modmachine.c

```diff
@@ -37,7 +37,9 @@
 
 #if MICROPY_PY_MACHINE
 
+#ifndef PAGE_SIZE
 #define PAGE_SIZE 4096
+#endif
```

Comment:
I don't know if this is the correct way to do it.  PAGE_SIZE may already be defined, but it may not be defined to the correct thing.

I think the right way to do it is use getpagesize(), which is supposed to give you the page size used by mmap.